azure-kusto-data/azure/kusto/data/request.py

```diff
@@ -49,6 +50,8 @@ def parse(cls, key):
                 return cls.application_certificate_thumbprint
             if key in ["authority id", "authorityid", "authority", "tenantid", "tenant", "tid"]:
                 return cls.authority_id
+            if key in ["aad federated security", "federated security", "federated", "fed", "aadfed"]:
+                return cls.aad_federated_security
             raise KeyError(key)
 
     def __init__(self, connection_string):

```

Comment:
I think we should explicitly expose `**kwargs` here and allow a `auth=False` to be set. would make code less cumbersome IMHO

Reply:
Discussed F2F, we are aligned in out connection strings in all languages.